### PR TITLE
Remove unneeded error check in ui tests

### LINKED ISSUE #48550: Most of UI tests are now compiled twice, once with `--error-format json` and once with `--error-format human`
The change was made in https://github.com/rust-lang/rust/pull/48337
cc @GuillaumeGomez 

This is not helpful for our testing times, especially given that number of UI tests will grow due to other test categories being converted to UI.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #48550.

cc @petrochenkov 

r? @Mark-Simulacrum

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
And it also solves the double compilation issue.

**Comment by Mark-Simulacrum** [comment]
I don't feel qualified to understand the check we're removing and why it's not necessary.

r? @petrochenkov

**Comment by petrochenkov** [comment]
You can't remove this, UI tests should check `ERROR`/`WARN`/etc annotations.
See https://github.com/rust-lang/rust/pull/46116 and linked issues for the background.

**Comment by GuillaumeGomez** [comment]
Even with the background, I don't understand its purpose. For this we typically had `compile-fail` tests. What's the point to check the compiler errors if you compare outputs? If an error/warning/help/whatever isn't thrown out by the compiler, the test will fail.

The other solution which I quickly started would be to actually parse the compiler stderr. And I can assure you that this is not gonna be fun or efficient...

**Comment by petrochenkov** [comment]
@GuillaumeGomez 
>Even with the background, I don't understand its purpose.

TLDR: UI testing alone is very unreliable.
Suppose you have a test with N lines that should report an error and M lines that should not report an error. How do you ensure this by having only the auto-generated `.stderr` file? "Manually" looking through two files trying to find corresponding lines is not reliable, it's very easy to miss something, it's like separating `assert`s from run-pass tests into separate files. `ERROR`/`WARN` annotations must be inline.

>For this we typically had compile-fail tests. 
The plan is to gradually migrate them to UI to consistently detect changes in secondary stuff like spans in addition to `ERROR` annotations (https://github.com/rust-lang/rust/issues/44844).

@estebank 
>I thought that ui tests were using the rendered field in the json output...

This was the case until https://github.com/rust-lang/rust/pull/48337 got merged.

**Comment by petrochenkov** [comment]
@GuillaumeGomez 
I see two ways to resolve https://github.com/rust-lang/rust/issues/48550:
- The simpler one: Revert https://github.com/rust-lang/rust/pull/48337/commits/1dc2015a9d71f3ef8191f6e0ca39b9154c87adfe, figure out the reason why the "for more information try `rustc --explain E0XXX`" stuff doesn't appear in text converted from json and fix it. If the fix is somehow fundamentally impossible, then test the "explain" message in a couple of dedicated tests using `--error-format=human` and leave all the remaining tests in pre-https://github.com/rust-lang/rust/pull/48337 state (this also means 1300 less lines of text in the repo, yay).
- The more complex, but maybe ultimately better one: Make `// ~ERROR`/`// ~WARN` checking work on `--error-format=human` output. This is what @nikomatsakis proposed originally (https://internals.rust-lang.org/t/consolidating-compile-fail-and-ui-testing-and-possibly-others-as-well/4896).

**Comment by GuillaumeGomez** [comment]
Then I'll just go for the `--error-format=human` output checking. I had someone working in most cases so I think it's doable.

**Comment by petrochenkov** [comment]
Closing in favor of https://github.com/rust-lang/rust/pull/48684
